openlibrary/plugins/worksearch/schemes/works.py

```diff
@@ -191,7 +191,11 @@ class WorkSearchScheme(SearchScheme):
     }
 
     def is_search_field(self, field: str):
-        return super().is_search_field(field) or field.startswith('id_')
+        # New variable introduced to prevent rewriting the input.
+        a_field = field
+        if a_field.startswith("work."):
+            a_field = a_field.partition(".")[2]
+        return super().is_search_field(a_field) or a_field.startswith('id_')
```

Comment:
This will let us avoid the new variable assignment and makes the code a little more state-less!

```suggestion
        if field.startswith("work."):
            return self.is_search_field(field.partition(".")[2])

        return super().is_search_field(field) or field.startswith('id_')
```